File: P2/Napakalaki/src/napakalaki/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;
import java.util.ArrayList;

/**
 *
 * @author Victor Vallecillo Morilla
 */
public class BadConsequence
{
    private String text; // para representar lo que dice un mal rollo
    private int levels, nVisibleTreasures, nHiddenTreasures;  
    /*   levels, de tipo int, para representar los niveles que se pierden.
        • nVisibleTreasures, de tipo int, para representar el número de tesoros visibles que se pierden.
        • nHiddenTreasures, de tipo int, para representar el número de tesoros ocultos que se pierden. */
    private boolean death;   // para representar un mal rollo de tipo muerte
    private ArrayList<TreasureKind> specificHiddenTreasures = new ArrayList();  // vector dinamico
    private ArrayList<TreasureKind> specificVisibleTreasures = new ArrayList();
    static final int MAXTREASURES = 10; //final indica que es de tipo constante.
    
    
    public BadConsequence(String text, int levels, int nVisible, int nHidden)
    {
        this.text=text;
        this.levels=levels;
        this.nVisibleTreasures=nVisible;
        this.nHiddenTreasures=nHidden;        
    }
    public BadConsequence(String text, boolean death)
    {
        this.text=text;
        this.death=death;
    }
    public BadConsequence(String text, int levels, ArrayList<TreasureKind> tVisible,
            ArrayList<TreasureKind> tHidden)
    {
        this.text=text;
        this.levels=levels;
        this.specificVisibleTreasures=tVisible; 
        this.specificHiddenTreasures=tHidden;
    }
    
    public ArrayList<TreasureKind> getSpecificHiddenTreasures(){
        return specificHiddenTreasures;
    }
    
    public ArrayList<TreasureKind> getSpecificVisibleTreasures(){
        return specificVisibleTreasures;
    }    
    
    public String getText()
    {
        return text;
    }
    public int getLevels()
    {
        return levels;
    }
    public int getnVisibleTreasures()
    {
        return nVisibleTreasures;
    }
    public int getnHiddenTreasures()
    {
        return nHiddenTreasures;
    }
    public boolean getDeath()
    {
        return death;
    }
    
    public String toString(){
        return "Text = " + text +
               " Levels = " + Integer.toString(levels) +
               " vTreasures = " + Integer.toString(nVisibleTreasures) +
               " hTreasures = " + Integer.toString(nHiddenTreasures) +
               " Death = " + Boolean.toString(death);
    }
    
    public boolean isEmpty()
    {
      boolean empty=false;
      if(levels==0 && death==false && nVisibleTreasures==0 && nHiddenTreasures==0 && 
              specificVisibleTreasures.size()==0 && specificHiddenTreasures.size()==0)
          empty=true;
      return empty;
    }
   
    public void substractVisibleTreasure(Treasure t)
    {
        
    }
    public void substractHiddenTreasure(Treasure t)
    {
        
    }
    
    public BadConsequence adjustToFitTreasureLists(Treasure [] v ,Treasure [] h )
    {
        return null;
    }
    
    
}

File: P2/NapakalakiRuby/lib/bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
# encoding: UTF-8
require_relative 'treasure_kind.rb'

class BadConsequence
  #con esto tenemos el get 
attr_reader :text, :levels, :nVisibleTreasures, :nHiddenTreasures, 
  :specificVisibleTreasures , :specificHiddenTreasures,:death 

  @@MAXTREASURES=10;
  
  def initialize(aText, someLevels, someVisibleTreasures, someHiddenTreasures, 
someSpecificVisibleTreasures, someSpecificHiddenTreasures, death)
    @text=aText
    @levels=someLevels
    @nVisibleTreasures=someVisibleTreasures
    @nHiddenTreasures=someHiddenTreasures
    @death=death
    @specificVisibleTreasures=someSpecificVisibleTreasures
    @specificHiddenTreasures=someSpecificHiddenTreasures
  end
private_class_method :new    
  #cn el self dice que es un metodo de la clase, como es un constructor, solo los constructores son de clase
  def self.newLevelNumberOfTreasures (aText, someLevels, someVisibleTreasures, someHiddenTreasures)
    new(aText, someLevels, someVisibleTreasures, someHiddenTreasures)
  end  
  def self.newLevelSpecificTreasures (aText, someLevels, someSpecificVisibleTreasures, someSpecificHiddenTreasures)
    new(aText,someLevels,0, 0, someSpecificVisibleTreasures, someSpecificHiddenTreasures)  
  end
  def self.newDeath (aText)
    new(aText, 0, 0, 0, Array.new, Array.new, true)  
  end
   
   def to_s
      put_s"Texto: #{@text} \n Niveles: #{@levels} \n Tesoros Visibles: #{@nVisibleTreasures} \n  Tesoros Ocutos: #{@nHiddenTreasures} 
      \n Tesoros Visibles Especificos #{@specificVisibleTreasures} \n Tesoros Ocultos Especificos #{@specificHiddenTreasures}
     \n Muerte #{@death}"
      
    end
    
   def isEmpty
     empty = false      
        if @levels == 0 && @death == false && @nHiddenTreasures == 0 && @nVisibleTreasures == 0  && 
            @specificHiddenTreasures.empty? && @specificVisibleTreasures.empty? then
            empty = true
        end
        
        return empty
   end
   
   def substractVisibleTreasure(t)
     
   end
   
   def substractHiddenTreasure(t)
     
   end
    
   def adjustToFitTreasureLists(v, h)
     
   end
   
  def to_s
    "text = #{@text} levels = #{@levels}"+
      " vTreasures = #{@nVisibleTreasures} hTreasures = #{@nHiddenTreasures}" +
      " sHTreasures = #{@specificHiddenTreasures} sVTreasures = #{@specificVisibleTreasures}"
  end
  
end
